src/Compilers/CSharp/Test/Semantic/Semantics/InitOnlyMemberTests.cs

```diff
@@ -0,0 +1,2255 @@
+                // (8,9): error CS8802: Init-only member 'C.Property' can only be assigned from a constructor, object initialization or 'with' expression of that type.
+                //         Property += 42; // 1
+                Diagnostic(ErrorCode.ERR_AssignmentInitOnly, "Property").WithArguments("C.Property").WithLocation(8, 9)
+                );
+        }
+
+        [Fact]
+        public void CompoundAssignmentDisallowed_OrAssignment()
+        {
+            string source = @"
+public class C
+{
+    public bool Property { get; init; }
+
+    void M()
+    {
+        Property |= true; // 1
+    }
+
+    C()
+    {
+        Property |= true;
+    }
+}
+";
+            var comp = CreateCompilation(new[] { source, IsExternalInitTypeDefinition }, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(
+                // (8,9): error CS8802: Init-only member 'C.Property' can only be assigned from a constructor, object initialization or 'with' expression of that type.
+                //         Property |= true; // 1
+                Diagnostic(ErrorCode.ERR_AssignmentInitOnly, "Property").WithArguments("C.Property").WithLocation(8, 9)
+                );
+        }
+
+        [Fact]
+        public void CompoundAssignmentDisallowed_NullCoalescingAssignment()
+        {
+            string source = @"
+public class C
+{
+    public string Property { get; init; }
+
+    void M()
+    {
+        Property ??= null; // 1
+    }
+
+    C()
+    {
+        Property ??= null;
+    }
+}
+";
+            var comp = CreateCompilation(new[] { source, IsExternalInitTypeDefinition }, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(
+                // (8,9): error CS8802: Init-only member 'C.Property' can only be assigned from a constructor, object initialization or 'with' expression of that type.
+                //         Property ??= null; // 1
+                Diagnostic(ErrorCode.ERR_AssignmentInitOnly, "Property").WithArguments("C.Property").WithLocation(8, 9)
+                );
+        }
+
+        [Fact]
+        public void CompoundAssignmentDisallowed_Increment()
+        {
+            string source = @"
+public class C
+{
+    public int Property { get; init; }
+
+    void M()
+    {
+        Property++; // 1
+    }
+
+    C()
+    {
+        Property++;
+    }
+}
+";
+            var comp = CreateCompilation(new[] { source, IsExternalInitTypeDefinition }, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(
+                // (8,9): error CS8802: Init-only member 'C.Property' can only be assigned from a constructor, object initialization or 'with' expression of that type.
+                //         Property++; // 1
+                Diagnostic(ErrorCode.ERR_AssignmentInitOnly, "Property").WithArguments("C.Property").WithLocation(8, 9)
+                );
+        }
+
+        [Fact]
+        public void RefProperty()
```

Comment:
>RefProperty [](start = 20, length = 11)

Please also test with non-auto property

Reply:
Also, please test what happens if we import a property like that from metadata. What symbols are we going to end up with. What happens if we try to consume it.

---
In reply to: [415201894](https://github.com/dotnet/roslyn/pull/43275#discussion_r415201894) [](ancestors = 415201894)